firmware/PinI2C_firmware.h

```diff
@@ -77,11 +77,11 @@
         }
         break;
 
-#if defined(ARDUINO_ARCH_AVR) // ESPs don't have analogReference()
+#if defined(ARDUINO_ARCH_AVR) || defined(ARDUINO_ARCH_SAMD) // ESPs don't have analogReference()
       case pinAnalogReferenceCmd: {
           if (i == 1) { //1 uint8_t
             uint8_t mode; bufferIn->read(mode);
-            analogReference(mode);
+            analogReference(static_cast<eAnalogReference>(mode));
```

Comment:
please see change in method signature. The compiler was complaining about the wrong type.

Reply:
Never heard of that enum, but makes sense. Would be nice if the Arduino docs would properly document each functions' parameter types, as it's not mentioned at https://www.arduino.cc/reference/en/language/functions/analog-io/analogreference/ and is hard to google.

Reply:
And frankly I don't understand why it compiles for ESPs as I can only find eAnalogReference used in the samd core:

```
~/.arduino15/packages$ find . -type f -exec grep -H --color eAnalogReference '{}' \;
./arduino/hardware/samd/1.8.13/cores/arduino/wiring_analog.c:void analogReference(eAnalogReference mode)
./arduino/hardware/samd/1.8.13/cores/arduino/Arduino.h:typedef enum _eAnalogReference
./arduino/hardware/samd/1.8.13/cores/arduino/Arduino.h:} eAnalogReference ;

```

Reply:
Damn, I thought I tested for Uno/avr's. What do you think is the best strategy? I think this caused a compile fail for samd.